bigtable/client/client_options_test.cc

```diff
@@ -46,3 +46,89 @@ TEST(ClientOptionsTest, EditCredentials) {
   EXPECT_EQ(typeid(grpc::InsecureChannelCredentials()),
             typeid(client_options_object.credentials()));
 }
+
+TEST(ClientOptionsTest, SetGrpclbFallbackTimeout) {
+  bigtable::ClientOptions client_options_object = bigtable::ClientOptions();
+  client_options_object.SetGrpclbFallbackTimeout(5);
+  grpc::ChannelArguments c_args = client_options_object.channel_arguments();
+  grpc_channel_args test_args = c_args.c_channel_args();
+  EXPECT_EQ(static_cast<size_t>(2), test_args.num_args);
+  // Because no accessors use low level C API.
+  EXPECT_EQ(GRPC_ARG_GRPCLB_FALLBACK_TIMEOUT_MS,
+            grpc::string(test_args.args[1].key));
+}
+
+TEST(ClientOptionsTest, SetCompressionAlgorithm) {
+  bigtable::ClientOptions client_options_object = bigtable::ClientOptions();
+  client_options_object.SetCompressionAlgorithm(GRPC_COMPRESS_NONE);
+  grpc::ChannelArguments c_args = client_options_object.channel_arguments();
+  grpc_channel_args test_args = c_args.c_channel_args();
+  EXPECT_EQ(static_cast<size_t>(2), test_args.num_args);
+  // Because no accessors use low level C API.
+  EXPECT_EQ(GRPC_COMPRESSION_CHANNEL_DEFAULT_ALGORITHM,
```

Comment:
Just curious, but how is it that one knows that this is `args[1]` vs. `args[0]`?

Reply:
These setters insert at the back of args_ vector according to the code available.

Reply:
Okay, but only one the tests uses `args[0]`, and the rest use `args[1]`.  Why?  Should we document in the tests why that is?

Reply:
The default ChannelArgument constructor create a channel argument object by adding a default user agent prefix. Now, in most of the setter we are adding new arguments so the index 1 but in one setter which is SetUserAgentPrefix the setter appends our prefix to the exist one and doesn't add any other argument so the index 0. If you want me to document this I can do this.

Reply:
Good research, please include a brief comment saying that in the tests.  At this point the code is right, documenting why is the only thing left in my view.

Reply:
Do we really want to expose channel_arguments implementation to users?

Reply:
No, I meant this should be documented in the **tests**.  The next developer that goes through the test code might wonder about it too.